sdks/java/harness/src/test/java/org/apache/beam/fn/harness/BeamFnDataReadRunnerTest.java

```diff
@@ -352,6 +354,35 @@ public void testRegistration() {
       fail("Expected registrar not found.");
     }
 
+    @Test
+    public void testSplittingBeforeStartBundle() throws Exception {
+      List<WindowedValue<String>> outputValues = new ArrayList<>();
+      BeamFnDataReadRunner<String> readRunner =
+          createReadRunner(outputValues::add, PTRANSFORM_ID, mockBeamFnDataClient);
+      // The split should happen at 5 since the allowedSplitPoints is empty.
+      assertEquals(
```

Comment:
A bundle can be started before each transforms `start` method has been invoked since we invoke `start` starting from the leaves and going to the parents and then we invoke `finish` in reverse order.

Having this definition of `started` allows us to get msec counters for `start` methods since they can be slow, checkpointing without processing through this read transform...

Reply:
I see. So that means we are now allowing splitting before processing gets started, right?

Reply:
We always did before as well since `registerInputLocation` doesn't mean we have seen any elements yet just that we are eligible to get them. This expands that time frame before receiving the first element slightly.